### PR TITLE
Fix win_user when specifying groups

### PR DESCRIPTION
**Pull request by trois-six**
win_user wasn't working here.

The GetType().Name for a string is "string[]" and not "string".
If $group_obj is null , a condition on $group_obj.GetType fails.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @Trois-Six. @angstwad @cchurch please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: windows/win_user.ps1 [hunk 1]
```diff
@@ -101,7 +101,8 @@ If ($account_locked -ne $null) {
 
 $groups = Get-Attr $params "groups" $null
 If ($groups -ne $null) {
-    If ($groups.GetType().Name -eq "String") {
+    $groups_type = $groups.GetType().Name
+    If ($groups_type -eq "String[]" -or $groups_type -eq "String") {
         [string[]]$groups = $groups.Split(",")
```
Comment (jhawkesworth) [inline windows/win_user.ps1:106]: I haven't looked at the rest of this but will this line work if $groups_type is "String[]" or "String"?.
If $groups_type can only be "String[]" then the -or $groups_type -eq "String" is not needed perhaps?

Comment (trois-six) [inline windows/win_user.ps1:106]: I don't have my env to test now, but I have seen that sometimes, the GetType().name is String and sometimes it's String[].

I have to re-do it, but basically there are four cases :
- name: Set Windows users
  win_user: name=mylogin password=mypassword groups_action=replace groups=["Users", "Administrators"] state=present
- name: Set Windows users
  win_user: name=mylogin password=mypassword groups_action=replace groups=["Users"] state=present
- name: Set Windows users
  win_user: name=mylogin password=mypassword groups_action=replace groups=Users,Administrators state=present
- name: Set Windows users
  win_user: name=mylogin password=mypassword groups_action=replace groups=Users state=present

**Comment by robynbergeron** [comment]
@angstwad @cchurch This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
@angstwad @cchurch still waiting on your review.  Please comment with text 'shipit' or 'needs_revision' as appropriate. If we don't hear from you within 14 days, we will start to look for additional maintainers for this module.

**Comment by gregdek** [comment]
@cchurch ping?

**Comment by robynbergeron** [comment]
@nitzmahone This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
@nitzmahone still waiting on your review.  Please comment with text 'shipit' or 'needs_revision' as appropriate. If we don't hear from you within 14 days, we will start to look for additional maintainers for this module.

**Comment by nitzmahone** [comment]
Curious if you're still seeing this issue- I'm not able to repro on current devel.

**Comment by gregdek** [comment]
@Trois-Six A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by nitzmahone** [comment]
Ended up fixing this in https://github.com/ansible/ansible-modules-core/commit/2f46c35f471b18f369c357016d0e430a65f8429b and backporting to stable-2.0. There were a whole raft of type checking issues in this module, and they were exacerbated by the StrictMode changes we did for 2.0. Should be all good now!

**Pull request closed without merging**